### PR TITLE
feat: implement cached document loader for improved external URL

### PR DESCRIPTION
**Pull request by kinxa0**
- added cached document loader for caching context url

### CODE REVIEW

File: src/utils/CachedDocumentLoader.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+export const buildCachedDocumentLoader = (logger: TsLogger): DocumentLoaderWithContext => {
+  // Return a factory function that Credo calls with agentContext
+  return (agentContext: AgentContext) => {
+    // Get the default Credo loader for this context
+    // handles: bundled DEFAULT_CONTEXTS, DID resolution, basic HTTP
+    const defaultLoader = defaultDocumentLoader(agentContext)
+
+    return async (url: string): Promise<DocumentLoaderResult> => {
+      // Let Credo handle bundled contexts and DIDs natively
+      const isExternalUrl = url.startsWith('http://') || url.startsWith('https://')
```
Comment (devdgna) [inline src/utils/CachedDocumentLoader.ts]: This check runs before `defaultDocumentLoader`, so it also intercepts Credo’s bundled HTTP(S) JSON-LD contexts, for example `https://www.w3.org/2018/credentials/v1`. That means a stale or poisoned `jsonld:document:*` cache entry can override Credo’s trusted `DEFAULT_CONTEXTS` and change JSON-LD canonicalization / proof verification behavior.

Can we make the cache skip bundled contexts, or otherwise let Credo’s built-in context handling run before consulting Redis/in-memory cache? The cache should only apply to documents Credo would actually fetch externally.
Acho @kinxa0

File: src/utils/CachedDocumentLoader.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import type { DocumentLoaderWithContext } from '@credo-ts/core/build/modules/vc/data-integrity/libraries/documentLoader'
+import type { DocumentLoaderResult } from '@credo-ts/core/build/modules/vc/data-integrity/libraries/jsonld'
+
+import { CacheModuleConfig } from '@credo-ts/core'
+import { defaultDocumentLoader } from '@credo-ts/core/build/modules/vc/data-integrity/libraries/documentLoader'
+
+const DOCUMENT_LOADER_CACHE_PREFIX = 'jsonld:document:'
+const DOCUMENT_CACHE_TTL_SECONDS = 86400 // 24 hours
+
+export const buildCachedDocumentLoader = (logger: TsLogger): DocumentLoaderWithContext => {
```
Comment (devdgna) [inline src/utils/CachedDocumentLoader.ts:15]: Can we add focused tests for the cache wrapper before merging? At minimum: cache hit returns without calling the default loader, cache miss fetches then stores with the expected TTL, and cache get/set failures fall back to the default loader. This code sits in the W3C verification/signing path, so regressions here would be hard to diagnose from API behavior alone.

**Review by devdgna — CHANGES_REQUESTED** [review]

**Comment by kinxa0** [comment]
Upon doing some research, it seems the credo already has a bundled context urls so these changes are pointless since it already reads from the memory instead of http call. @devdgna

**Comment by kinxa0** [comment]
Reopening for caching schema context url

**Review by devdgna — APPROVED** [review]
LGTM
